package.json

```diff
@@ -626,6 +626,13 @@
                 "category": "Jupyter",
                 "enablement": "isWorkspaceTrusted && (jupyter.interactive.canRestartNotebookKernel || (notebookKernel =~ /^ms-toolsai.jupyter\\// && jupyter.notebookeditor.canrestartNotebookkernel))"
             },
+            {
+                "command": "jupyter.restartkernelandrunuptoselectedcell",
+                "title": "%jupyter.command.jupyter.restartkernelandrunuptoselectedcell.title%",
+                "shortTitle": "%jupyter.command.jupyter.restartkernelandrunuptoselectedcell.title%",
+                "category": "Jupyter",
+                "enablement": "isWorkspaceTrusted && (jupyter.interactive.canRestartNotebookKernel || (notebookKernel =~ /^ms-toolsai.jupyter\\// && jupyter.notebookeditor.canrestartNotebookkernel))"
```

Comment:
Wait, why are we adding this new command, i thought this was `Restart and Run all`, this seems to be `Restart and Run upto selected cell`
Why do we need this new command, i didn't realize this new command was a requirement (also i do not think this is something that exists in Jupyter either, hence feels a little odd to add this)

Reply:
<img width="759" alt="image" src="https://user-images.githubusercontent.com/876920/233141267-617e99e7-80e4-4dee-ae08-fa429fa09fa2.png">

@DonJayamanne it exists in Jupyter Lab.